update-versions.sh

```diff
@@ -18,55 +19,89 @@ fi
 
 set -ex
 
+if ! which rg &> /dev/null; then
+    echo "Command `rg` (ripgrep) was not found. Please install it and try again."
+    exit 1
+fi
+
+if ! cargo set-version --help &> /dev/null; then
+    echo "Cargo extension `cargo-edit` is not installed. Please install it by running: cargo install cargo-edit"
+    exit 1
+fi
+
 HEAD=$(git rev-parse HEAD)
 VERSION=$1
 
-CARGO_TOMLS_TO_BUMP=(
-    ./Cargo.toml
-    ./pgx/Cargo.toml
-    ./pgx-utils/Cargo.toml
-    ./pgx-macros/Cargo.toml
-    ./pgx-tests/Cargo.toml
-    ./cargo-pgx/Cargo.toml
-    ./pgx-pg-sys/Cargo.toml
-)
-
-CARGO_TOMLS_TO_SED=(
-    ./cargo-pgx/src/templates/cargo_toml
-    ./nix/templates/default/Cargo.toml
-    ./pgx/Cargo.toml
-    ./pgx-utils/Cargo.toml
-    ./pgx-macros/Cargo.toml
-    ./pgx-tests/Cargo.toml
-    ./cargo-pgx/Cargo.toml
-    ./pgx-pg-sys/Cargo.toml
-    ./pgx-examples/*/Cargo.toml
-    ./Cargo.toml
-)
-
-DEPENDENCIES_TO_UPDATE=(
-    "pgx"
-    "pgx-tests"
-    "pgx-macros"
-    "pgx-pg-config"
-    "pgx-pgx-sys"
-    "pgx-utils"
-    "cargo-pgx"
-)
-
-SEMVER_REGEX="(?P<major>0|[1-9]\d*)\.(?P<minor>0|[1-9]\d*)\.(?P<patch>0|[1-9]\d*)(?:-(?P<prerelease>(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+(?P<buildmetadata>[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?"
-
-for cargo_toml in ${CARGO_TOMLS_TO_SED[@]}; do
-    for dependency in ${DEPENDENCIES_TO_UPDATE[@]}; do
-        rg --passthru -N "(?P<prefix>^${dependency}.*\")(?P<pin>=?)${SEMVER_REGEX}(?P<postfix>\".*$)" -r "\${prefix}=${VERSION}\${postfix}" ${cargo_toml} > ${cargo_toml}.tmp || true
+# Use `cargo set-version` to update all main project files
+function update_main_files() {
+    local version=$1
+
+    EXCLUDE_PACKAGES=()
+
+    # Add additional packages to ignore by using the following syntax:
+    # EXCLUDE_PACKAGES+=('foo' 'bar' 'baz')
+
+    # Ignore all packages in pgx-examples/
+    for file in $(find pgx-examples/ -type f -name "Cargo.toml"); do
+        EXCLUDE_PACKAGES+=($(cat $file | rg --multiline '\[package\](.*\n)(?:[^\[]*\n)*name\s?=\s?"(?P<name>[-_a-z]*)"(.*\n)*' -r "\${name}"))
+    done
+
+    echo "Excluding the following packages:"
+    for p in "${EXCLUDE_PACKAGES[@]}"; do
+        echo " * $p"
+    done
+    echo ""
+
+    cargo set-version $(echo "${EXCLUDE_PACKAGES[@]/#/--exclude }") --workspace $version
+}
+
+# This is a legacy holdover for updating extra toml files throughout various crates
+function update_extras() {
+    local version=$1
+
+    # ordered in a topological fashion starting from the workspace root Cargo.toml
+    # this isn't always necessary, but it's nice to use a mostly-consistent ordering
+    CARGO_TOMLS_TO_BUMP=(
+        ./Cargo.toml
+    )
+
+    CARGO_TOMLS_TO_SED=(
+        ./Cargo.toml
+        ./pgx-examples/*/Cargo.toml
+        ./cargo-pgx/src/templates/cargo_toml
+        ./nix/templates/default/Cargo.toml
```

Comment:
```suggestion
        ./cargo-pgx/src/templates/cargo_toml
        ./nix/templates/default/Cargo.toml
```

Maybe we should invert `./nix/templates` into `./templates/nix/` and have `./cargo-pgx/src/templates` be its own template as well, and include them properly in the workspace and all that. Future PR, though, since now the release script at least works.

Reply:
Very possible, though I am not too familiar with nix and any repercussions of moving the nix template file somewhere else.